### PR TITLE
offload_checksum_windows.py: add windows support for Offload Tx/Rx checksum

### PR DESCRIPTION
**Pull request by kimi1978wy**
Add case of Offload Tx/Rx checksum for windows guests

Signed-off-by: Yu Wang <wyu@redhat.com>

id:1566941

### CODE REVIEW

**Comment by kimi1978wy** [comment]
(1/1) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.offload_checksum_windows: PASS (225.84 s)
 (1/1) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.offload_checksum_windows: PASS (391.49 s)

**Comment by kimi1978wy** [comment]
depends on https://github.com/autotest/tp-qemu/pull/1829

File: qemu/tests/offload_checksum_windows.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    3) restart nic, and run file transfer test
+    4) set the tx/rx offload checksum of the nekvm driver to disable
+    5) restart nic, and run file transfer test again
+
+    param test: the test object
+    param params: the test params
+    param env: test environment
+    """
+
+    def set_offload_checksum_windows(vm, is_tx, type):
```
Comment (PaulYuuu) [inline qemu/tests/offload_checksum_windows.py]: Please do not use `type` as a keyword.

File: qemu/tests/offload_checksum_windows.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            param = "Offload.TxChecksum"
+        else:
+            param = "Offload.RXCS"
+        if type == "tcp":
+            value = "1"
+        else:
+            value = "0"
+        utils_net.set_netkvm_param_value(vm, param, value)
+
+    def start_test(type="tcp"):
```
Comment (PaulYuuu) [inline qemu/tests/offload_checksum_windows.py]: Same as above.

Comment (kimi1978wy) [reply qemu/tests/offload_checksum_windows.py]: Modified, thanks for reminder :)

File: qemu/tests/offload_checksum_windows.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        on windows.
+
+        param vm: the target vm
+        param is_tx: True for tx setting, False for rx setting
+        param checksum_config: the config for checksum settings, one of 'tcp' or 'disable'
+        """
+        if is_tx:
+            param = "Offload.TxChecksum"
+        else:
+            param = "Offload.RXCS"
```
Comment (PaulYuuu) [inline qemu/tests/offload_checksum_windows.py]: Suggest use a dict like:
`offload_param = {'tx': 'Offload.TxChecksum', 'rx': 'Offload.RXCS'}`

File: qemu/tests/offload_checksum_windows.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        if checksum_config == "tcp":
+            value = "1"
+        else:
+            value = "0"
```
Comment (PaulYuuu) [inline qemu/tests/offload_checksum_windows.py]: If only 0 and 1,  I suggest use `int(True/False)`, or `int(bool(str/None))`

File: qemu/tests/offload_checksum_windows.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    timeout = params.get("timeout", 360)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    session = vm.wait_for_login(timeout=timeout)
+    error_context.context("Check if the driver is installed and "
+                          "verified", logging.info)
+    driver_name = params.get("driver_name", "netkvm")
+    session = utils_test.qemu.windrv_check_running_verifier(session, vm,
```
Comment (PaulYuuu) [inline qemu/tests/offload_checksum_windows.py]: In fact, if there is a 'vm' parameter, I don't think it's necessary to add `session', but this is another problem, don't mind.

File: qemu/tests/offload_checksum_windows.py (hunk 3 above)
Comment (kimi1978wy) [reply qemu/tests/offload_checksum_windows.py]: is_tx is a boolean value, so it is not necessary to build a dict for these value. I'll make it a single line like this:
`offload_param = "offload.TxChecksum" if is_tx else "offload.RXCS"`

File: qemu/tests/offload_checksum_windows.py (hunk 4 above)
Comment (kimi1978wy) [reply qemu/tests/offload_checksum_windows.py]: Same reason as above, use single line `if else` to make it simpler.

File: qemu/tests/offload_checksum_windows.py (hunk 5 above)
Comment (kimi1978wy) [reply qemu/tests/offload_checksum_windows.py]: Yeah, you are right. I'll keep `session` because it's demanded by this method.

**Comment by kimi1978wy** [comment]
@PaulYuuu updated some codes, pls help to review it. Thanks a lot
(1/4) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.offload_checksum_windows.q35: PASS (448.42 s)
 (3/4) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.offload_checksum_windows.q35: PASS (565.12 s)

**Comment by PaulYuuu** [comment]
The code is ok for me.

**Pull request merged**